index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
Okay. Then I'll update the signature and make the other changes you requested @davidchambers.